pymeasure/instruments/tdk/tdk_base.py

```diff
@@ -0,0 +1,387 @@
+from pymeasure.instruments.validators import strict_discrete_range
+import logging
+from time import sleep
+import numpy as np
+
+# =============================================================================
+# Logging
+# =============================================================================
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+# =============================================================================
+# Instrument file
+# =============================================================================
+
+
+class TDK_Lambda_Base(Instrument):
+    """
+    This is the base class for TDK Lambda Genesys Series DC power supplies.
+
+    Do not directly instantiate an object with this class. Use one of the
+    TDK-Lambda power supply instrument classes that inherit from this parent
+    class.
+
+    Untested commands are noted in docstrings.
+    """
+
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    # Initializer and important communication methods
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    def __init__(self, adapter, name="TDK-Lambda Base", address=6, **kwargs):
+        """
+        The initialization of a TDK instrument requires the current address
+        of the TDK power supply. The default address for the TDK Lambda is 6.
+
+        :param adapter: VISAAdapter instance
+        :param name: Instrument name
+        :param address: Serial port daisy chain number. Default is 6.
+        """
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=False,
+            asrl={'read_termination': "\r", 'write_termination': "\r"},
+            **kwargs
+        )
+        self.address = address
+
+    def write(self, command):
+        """Replace the ``Instrument.write()`` method to strip out an "OK" reply
+        that the instrument returns for any non-querying commands.
+
+        By default, any non-querying commands (i.e., a command that does NOT
+        have the "?" symbol in it like the instrument command "PV 10") will
+        automatically return an "OK" reply. This is done to confirm that the
+        instrument has received the command. Any querying commands (i.e., a
+        command that does have the "?" symbol in it like the instrument command
+        "PV?") will return the requested value. The returned value itself is
+        confirmation that the command was received.
+
+        The default, the ``Instrument.write()`` method is not set up to
+        automatically strip out instrument confirmations. If the command
+        string does not contain a "?", then ``self.read()`` once to clear the
+        "OK" from the read buffer. If this is not done, the adapter read buffer
+        will hold the "OK" reply until the next read command is given.
+
+        :param command: Command string to be sent to the instrument.
+        """
```

Comment:
In the other cases with confirmations, we use check_errors to read after writing, for that reason I made #883 .

Reply:
I see what you mean, I thought of `check_errors` only for SCPI. I'll look into using that pattern, we would only `check_errors` for non-query commands.

Reply:
Ok, I have implemented and tested that, moved the write override logic to `check_errors` and utilized `check_set_errors` flag.